cluster_test.spec.ts

```diff
@@ -0,0 +1,165 @@
+  describe('Validating Incoming Twilio Request', () => {
+    let tunnel;
+    let flowSid;
+    let validationServer;
+    let portNumber = 7777;
+    let validationCount = 0; //workaround to ensure validation server receives requests (due to localtunnel issue)
+    beforeAll(async () => {
+      validationServer = await http.createServer((req, res) => {
+        validationCount++;
+        let url = req.headers["x-forwarded-proto"] + "://" + req.headers["host"] + req.url
+        let signatureHeader = req.headers["x-twilio-signature"]
+        let body = "";
+        req.on("data", (chunk: string) => {
+          body += chunk;
+        });
+
+        req.on("end", () => {
+          let params = new URLSearchParams(body);
+          let paramObject = Object.fromEntries(params.entries());
+          let requestIsValid = twilio.validateRequest(process.env.TWILIO_AUTH_TOKEN, signatureHeader, url, paramObject)
+          expect(requestIsValid).toBeTruthy();
+        });
+      });
+
+      validationServer.listen(portNumber);
+      tunnel = await localtunnel({port: portNumber});
+    });
+
+    afterAll(() => {
+      tunnel.close();
+      testClient.studio.v2.flows(flowSid).remove();
+      validationServer.close();
+    });
+
+    function CreateStudioFlow(url: string, method: string): Promise<FlowInstance> {
+      return testClient.studio.v2.flows.create({
+        friendlyName: 'Node Cluster Test Flow',
+        status: 'published',
+        definition: {
+          description: 'Studio Flow',
+          states: [
+            {
+              name: 'Trigger',
+              type: 'trigger',
+              transitions: [{
+                "next": "httpRequest",
+                "event": "incomingRequest"
+              }],
+              properties: {}
+            }, {
+              name: "httpRequest",
+              type: "make-http-request",
+              transitions: [],
+              properties: {
+                method: method,
+                content_type: "application/x-www-form-urlencoded;charset=utf-8",
+                url: url
+              }
+            }
+          ],
+          initial_state: 'Trigger',
+          flags: {
+            allow_concurrent_calls: true
+          }
+        }
+      })
+    }
+
+    async function validateRequest(method: string) {
+      let flow = await CreateStudioFlow(tunnel.url, method);
+      flowSid = flow.sid;
+      await testClient.studio.v2.flows(flowSid)
+        .executions
+        .create({to: 'to', from: 'from'});
+      await new Promise(resolve => setTimeout(resolve, 5000))
+    }
+
+    test("Should validate GET request", async () => {
+      await validateRequest('GET');
+      expect(validationCount).toBe(1);
+    })
+
+    test("Should validate POST request", async () => {
+      await validateRequest('POST');
+      expect(validationCount).toBe(2);
+    })
+  });
+})
+
+test("Should list available numbers", () => {
+  return testClient.availablePhoneNumbers('US')
```

Comment:
Should group this with the other non-webhook tests. Maybe move the webhook validation tests to a separate spec?